celery/backends/redis.py

```diff
@@ -99,6 +101,20 @@ def on_after_fork(self):
             logger.warning(text_t(e))
         super(ResultConsumer, self).on_after_fork()
 
+    def _reconnect_pubsub(self):
+        self._pubsub = None
+        self.backend.client.connection_pool.reset()
+        # task state might have changed when the connection was down so we
+        # retrieve meta for all subscribed tasks before going into pubsub mode
+        metas = self.backend.client.mget(self.subscribed_to)
```

Comment:
Can we use `mget` with channels?

Reply:
No we can't. This uses the fact that `RedisBackend.set` not only publishes the task status update to the channel, but also assigns it to a key with the same name. See https://github.com/celery/celery/blob/90fe53f901cf2e71c570fa639aa9b39ad228fadd/celery/backends/redis.py#L329-L333
In normal operation the consumer listens to messages published to the channel, but when reconnecting it will fetch all messages from the keys instead. This is tested since https://github.com/celery/celery/pull/5921/files#diff-5cb790f5c0c0aea16658640ccd102c19R246 is `RedisBackend.set` and not `Redis.set`.